### PR TITLE
#4290 add community officer role

### LINKED ISSUE #4290: Add a new community-level userrole with web access
### Situation Description

A lot of german users need a community-level user with access to the webapp.

### Feature Description

For now, add a new user role. I will complete this issue with the required userrights asap.

Required userrights:

- listing cases & contacts in their community
- creating and sending quarantine documents & subsequently change the related quarantine order date
- Editing the case-person and contact-person

Creating new Cases and Contacts does not seem to be a requirement.

### Possible Alternatives

Waiting for the new user rights model is not an option, due to this issues urgency.

### PR DESCRIPTION
**Pull request by DavidBaldsiefen**
Closes #4290.

The given rights are a combination of COMMUNITY_INFORMANT, CONTACT_OFFICER and SURVEILLANCE_OFFICER roles. 

The idea was to rather give this user more rights than necessary, instead of ending up with additional requests for another userrole with more rights. All this should no longer be necessary when the new user role system arrives.

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/user/UserRight.java [hunk 1]
```diff
@@ -578,7 +606,8 @@
 			DISTRICT_OBSERVER,
 			CASE_SUPERVISOR,
 			CONTACT_SUPERVISOR,
-			POE_SUPERVISOR
+			POE_SUPERVISOR,
+			COMMUNITY_OFFICER
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/user/UserRight.java:610]: Remove this user right, should only be available to supervisor roles and above

File: sormas-api/src/main/java/de/symeda/sormas/api/user/UserRight.java [hunk 2]
```diff
@@ -716,12 +754,15 @@
...
+			IMPORT_USER,
+			COMMUNITY_OFFICER
 	),
 	WEEKLYREPORT_CREATE(
 			HOSPITAL_INFORMANT,
 			COMMUNITY_INFORMANT,
-			SURVEILLANCE_OFFICER
+			SURVEILLANCE_OFFICER,
+			COMMUNITY_OFFICER,
+			COMMUNITY_OFFICER
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/user/UserRight.java:765]: Accidentally added twice here

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by MateStrysewske** [comment]
@DavidBaldsiefen Can you please resolve the merge conflicts before I test it in the UI?

**Comment by DavidBaldsiefen** [comment]
@MateStrysewske done

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Seems to work fine, the only thing I'd change is to give Community Officers access to the surveillance and contact dashboards (since they also have the respective view and edit rights)

**Comment by DavidBaldsiefen** [comment]
> Seems to work fine, the only thing I'd change is to give Community Officers access to the surveillance and contact dashboards (since they also have the respective view and edit rights)

I agree, but I will need to check if the dashboards are filtered correctly. I am not sure if that is the case at the moment, I will check it out tomorrow

**Comment by DavidBaldsiefen** [comment]
I checked it locally in the UI, looks good to me!

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**